File: src/main/scala/resonantengine/prefab/block/itemblock/ItemBlockMetadata.scala
package resonantengine.prefab.block.itemblock

import net.minecraft.block.Block
import net.minecraft.item.ItemStack
import resonantengine.lib.utility.LanguageUtility

class ItemBlockMetadata(block: Block) extends ItemBlockTooltip(block)
{
  setHasSubtypes(true)

  override def getMetadata(damage: Int): Int =
  {
    return damage
  }

  override def getUnlocalizedName(itemstack: ItemStack): String =
  {
    val localized: String = LanguageUtility.getLocal(getUnlocalizedName() + "." + itemstack.getItemDamage() + ".name")
    if (localized != null && !localized.isEmpty)
    {
      return getUnlocalizedName() + "." + itemstack.getItemDamage
    }
    return getUnlocalizedName()
  }
}

File: src/main/scala/resonant/content/prefab/itemblock/ItemBlockMetadata.java
package resonant.content.prefab.itemblock;

import net.minecraft.block.Block;
import net.minecraft.item.ItemStack;
import resonant.lib.utility.LanguageUtility;

public class ItemBlockMetadata extends ItemBlockTooltip
{
	public ItemBlockMetadata(Block block)
	{
		super(block);
		this.setHasSubtypes(true);
	}

	@Override
	public int getMetadata(int damage)
	{
		return damage;
	}

	@Override
	public String getUnlocalizedName(ItemStack itemstack)
	{
		String localized = LanguageUtility.getLocal(getUnlocalizedName() + "." + itemstack.getItemDamage() + ".name");
		if (localized != null && !localized.isEmpty())
		{
			return getUnlocalizedName() + "." + itemstack.getItemDamage();
		}
		return getUnlocalizedName();
	}
}
